### PR TITLE
Use utf16 encoding offsets for documents

### LINKED ISSUE #148: Illegal character error.
When I use ocaml-lsp with emacs, I have a following issue: when I accidentally type in Russian and then press backspace erasing all the visible characters, I still have invisible bytes left, leading to illegal character error.

Say, when I type "дуе" instead of "let", then press backspace three times, I'm getting three errors: `Illegal character (\131); Illegal character (\181); Illegal character (\180)`. ("дуе" is represented as "\208\180\209\131\208\181", so it seems like it deletes only first byte of each symbol.)

I don't have such an issue with neither other `lsp` implementations, nor `merlin`.

### PR DESCRIPTION
**Pull request by giltho**
Fixes #148

Probably also #161, we'll confirm with issue author

### CODE REVIEW

**Comment by rgrinberg** [comment]
Looks good, but you need to add uutf as a submodule and make sure that it builds with dune.

**Comment by giltho** [comment]
oh yeah I forgot
Can't I just add it to the dune-project as a dependency ? Do we need to vendor it ?

**Comment by rgrinberg** [comment]
Looks like there’s even an appropriate fork out there: https://github.com/dune-universe/uutf

We can’t add it as a normal dependency for a couple of reasons:

* A user might want to pick a different version of uutf for their project. This should be independent of the version of uutf their lsp server is using. It’s a bit unfortunate, but we’re just working around a limitation of opam
* We can’t use any non dune dependencies. The server should be buildable without installing ocamlbuild/findlib.

**Comment by giltho** [comment]
Ah very good points ! Never thought about that
Ok, on it then :)

**Comment by giltho** [comment]
Thanks for the link to the fork!

File: ocaml-lsp-server/src/document.ml [hunk 1]
```diff
@@ -27,7 +27,7 @@ module Syntax = struct
     | id -> failwith ("Unexpected language id " ^ id)
 end
 
-let { Logger.log } = Logger.for_section "ocaml-lsp-server"
+(* let { Logger.log } = Logger.for_section "ocaml-lsp-server" *)
```
Comment (rgrinberg) [inline ocaml-lsp-server/src/document.ml:30]: Why are we commenting this stuff out?

File: ocaml-lsp-server/src/document.ml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-open Import
+(* open Import *)
```
Comment (rgrinberg) [inline ocaml-lsp-server/src/document.ml:1]: this can be `open! Import` if it's causing warnings.

File: ocaml-lsp-server/src/document.ml (hunk 1 above)
Comment (giltho) [reply ocaml-lsp-server/src/document.ml:30]: Ah I forgot to clean that up
The logger is only used here for the logging of the whole new text every time a document changes. Which 90% of the "over-logging" we do.
I think the server is stable enough to remove this now.

Comment (rgrinberg) [reply ocaml-lsp-server/src/document.ml]: Yeah, I never found this message to be overly useful either.

**Pull request merged**